### PR TITLE
Basic force directed graph showing all paths

### PR DESCRIPTION
**Pull request by 123joshuawu**
Quick visualization of all paths using force directed graph

Adds `d3` dependency
Adds `PathPageQuery` for querying paths
Access `PathPage` via `/path`
Adds `ForceDirectedGraph` which is d3 force directed graph

Start #17

### CODE REVIEW

File: gui/src/ts/components/pages/PathPage.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import {Node} from "models/Node";
+
+interface PathNode extends d3.SimulationNodeDatum, Node {
+  pathId: string;
+}
+
+interface PathEdge extends d3.SimulationLinkDatum<PathNode> {}
+
+export const PathPage: React.FunctionComponent = () => {
+  const {data} = useQuery<PathPageQuery>(PathPageDocument);
```
Comment (gordom6) [inline gui/src/ts/components/pages/PathPage.tsx]: Don't forget to handle "error" in all of these top-level queries.

I'm going to file an issue to abstract out our error handling a little.

Comment (gordom6) [reply gui/src/ts/components/pages/PathPage.tsx]: I'll make the change here myself.

File: gui/src/ts/components/pages/PathPage.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    nodes: PathNode[];
+    links: PathEdge[];
+  }>(() => {
+    const nodes: PathNode[] = [];
+    const links: PathEdge[] = [];
+
+    data?.paths.forEach((path) => {
+      path.edges.forEach((edge) => {
+        if (
+          edge.subjectNode &&
```
Comment (gordom6) [inline gui/src/ts/components/pages/PathPage.tsx:36]: These should always exist but fine to check.

File: gui/src/ts/components/pages/PathPage.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          nodes.find((node) => node.id === edge.subject) === undefined
```
Comment (gordom6) [inline gui/src/ts/components/pages/PathPage.tsx:37]: Nit: prefer !nodes.some

File: gui/src/ts/components/pages/PathPage.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        });
+      });
+    });
+
+    return {nodes, links};
+  }, [data]);
+  // console.log(pathGraphData);
+
+  return (
+    <Frame>
```
Comment (gordom6) [inline gui/src/ts/components/pages/PathPage.tsx:61]: No loading state?

File: gui/src/ts/components/pages/PathPage.tsx (hunk 2 above)
Comment (123joshuawu) [reply gui/src/ts/components/pages/PathPage.tsx:36]: Okay wasn't sure

File: gui/src/ts/components/pages/PathPage.tsx (hunk 3 above)
Comment (123joshuawu) [reply gui/src/ts/components/pages/PathPage.tsx:37]: Right changed

File: gui/src/ts/components/pages/PathPage.tsx (hunk 1 above)
Comment (123joshuawu) [reply gui/src/ts/components/pages/PathPage.tsx]: Gotcha, I'll keep that in mind

File: gui/src/ts/components/pages/PathPage.tsx (hunk 4 above)
Comment (123joshuawu) [reply gui/src/ts/components/pages/PathPage.tsx:61]: Whoops added

**Pull request merged**